### DISCUSSION TITLE
Icons folder

### DISCUSSION THREAD

**Discussion by sajkofff** [rejetto/hfs#330] [Q&A]
Hello. How to change folder icons?

![Pic](https://github.com/rejetto/hfs/assets/48749271/e8990e91-d2e5-4d2b-a3e7-58a14f1ca7ac)

**Comment by rejetto** [comment]
at the moment it's technically possible, but not exactly simple.
- Go to Admin > Custom HTML
- in section "before header" enter this
```html
<script>
HFS.onEvent('entryIcon', x => x.entry.isFolder && HFS.hIcon('password'))
</script>
```

in this example i'm using a built-in icon "password", but you can enter any web address.

What kind of icon did you plan to use?

I see the links in your screenshot are gray. Did you customize the color?

**Comment by rejetto** [comment]
I made a plugin that allow you to customize the icon based on the file extension, but it doesn't support folders.
https://github.com/rejetto/file-icons
Admin > plugins > search online

Give it a try, and see if you like the way it works, and in case I will expand it to support folders.

**Comment by rejetto** [comment]
well, I decided to go head and expand the plugin, and now it supports folders too.
I hope it works well for you.

**Reply by SimonChiu930** [reply]
How add the folder on Ext. field?

**Reply by rejetto** [reply]
folders have no extension.
you have to use the "icon for folders" field.

**Reply by SimonChiu930** [reply]
Work, thanks

**Comment by sajkofff** [comment]
Hello. Yes, I changed the color and icons of file extensions. It only remained to change the folder icons so that they looked in the same style.
Thanks for the updated plugin.

**Comment by sajkofff** [comment]
Once the icons changed, they do not change anymore. I don’t understand, something is holding the icons that I installed for the first time. The second time they don't change. If you disable the plugin, the icons become default. HFS restarted.
I'm sorry. I am an amateur.

**Comment by sajkofff** [comment]
Seems to have figured it out. It's because of the browser. Everything is fine in other browser.

**Comment by rejetto** [comment]
what browser?
it could be a cache problem. Changing the browser you don't have the same cache, and the problem disappears.

Solution should be: hold SHIFT while you reload (do it in the previous browser)

**Comment by sajkofff** [comment]
Yes, I already cleared the cache in the browser.  Thank you very much. Everything is working
